lib/ansible/module_utils/network/tetration.py

```diff
@@ -0,0 +1,381 @@
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE
+# USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+import os
+import json
+from functools import partial
+from ansible.module_utils._text import to_native
+from ansible.module_utils.six import iteritems, iterkeys
+from ansible.module_utils._text import to_text
+
+try:
+    from tetpyclient import RestClient
+    HAS_TETRATION_CLIENT = True
+except ImportError:
+    HAS_TETRATION_CLIENT = False
+try:
+    import urllib3
+    # Disable SSL Warnings
+    urllib3.disable_warnings()
+except ImportError:
+    pass
+
+# defining tetration constants
+TETRATION_API_INVENTORY_TAG = '/inventory/tags'
+TETRATION_API_ROLE = '/roles'
+TETRATION_API_TENANT = '/vrfs'
+TETRATION_API_USER = '/users'
+TETRATION_API_SENSORS = '/sensors'
+TETRATION_API_INVENTORY_FILTER = '/filters/inventories'
+TETRATION_API_SCOPES = '/app_scopes'
+TETRATION_API_INTERFACE_INTENTS = '/inventory_config/interface_intents'
+TETRATION_API_AGENT_NAT_CONFIG = '/agentnatconfig'
+TETRATION_API_APPLICATIONS = '/applications'
+TETRATION_API_APPLICATION_POLICIES = '/policies'
+TETRATION_API_AGENT_CONFIG_PROFILES = '/inventory_config/profiles'
+TETRATION_API_AGENT_CONFIG_INTENTS = '/inventory_config/intents'
+TETRATION_COLUMN_NAMES = '/assets/cmdb/attributenames'
+TETRATION_API_EXT_ORCHESTRATORS = '/orchestrator'
+
+TETRATION_PROVIDER_SPEC = {
+    'server_endpoint': dict(type='str', required=True, aliases=['endpoint', 'host']),
+    'api_key': dict(type='str', required=True),
+    'api_secret': dict(type='str', required=True, no_log=True),
+    'verify': dict(type='bool', default=False),
+    'silent_ssl_warnings': dict(type='bool', default=True),
+    'timeout': dict(type='int', default=10),
+    'max_retries': dict(type='int', default=3),
+    'api_version': dict(type='str', default='v1')
+}
+
+
+class TetrationApiBase(object):
+    ''' Base class for implementing Tetration API '''
+    provider_spec = {'provider': dict(type='dict',
+                                      options=TETRATION_PROVIDER_SPEC)}
+
+    def __init__(self, provider, module):
+        if not HAS_TETRATION_CLIENT:
+            raise Exception('tetpyclient is required but does not appear '
+                            'to be installed.  It can be installed using the '
+                            'command `pip install tetpyclient`')
+        if len(provider.keys()) > 0:
+            for key in provider.keys():
+                if key not in TETRATION_PROVIDER_SPEC.keys():
+                    del provider[key]
+        for key, value in iteritems(TETRATION_PROVIDER_SPEC):
+            if key not in provider:
+                # apply default values from NIOS_PROVIDER_SPEC since we cannot just
```

Comment:
```suggestion
                # apply default values from TETRATION_PROVIDER_SPEC since we cannot just
```
Is this what you meant?